### PR TITLE
Add reminder emails functionality

### PR DESCRIPTION
**Pull request by jungwire**
### Description:

Introduces some changes to the notification email service by adding a functionality to send reminder emails for pending (negotiations that are under review) and stale (negotiations without ) to admins and representatives, respectively. The goal is to increase the responsiveness of biobanks/admins.

### Checklist:

_Make sure you tick all the boxes bellow if they are true or do not apply:_

- [x] I have performed a self review of my code
- [x] My code follows [Google Java Code style](https://google.github.io/styleguide/javaguide.html)
- [ ] I have made my code as simple as possible
- [x] I have added unit tests and the code coverage has not decreased
- [ ] I have updated the documentation in all relevant places

### CODE REVIEW

**Review by RadovanTomik — CHANGES_REQUESTED** [review]
why not just implement a nightly cron method that goes through all Negotiations create e.g. a week ago, looks at what resouces have had at least one status update and sends a notification to reps of those that didn't have an update?

File: src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java [hunk 1]
```diff
@@ -33,10 +38,29 @@
...
 
   @Value("${negotiator.frontend-url}")
   private String frontendUrl;
 
+  @Value("${reminder.trigger-duration-days:P14D}")
+  private String triggerDuration;
+
+  private static LocalDateTime thresholdTime;
+
+  private static Collection<NegotiationResourceState> activeNegotiationResourceStates =
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java:51]: why not do it the other way? nonActiveResource states are just Unavailable and Made available

File: src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java [hunk 2]
```diff
@@ -33,10 +38,29 @@ public class UserNotificationServiceImpl implements UserNotificationService {
   @Autowired EmailService emailService;
   @Autowired ResourceLifecycleService resourceLifecycleService;
   @Autowired TemplateEngine templateEngine;
+  @Autowired NegotiationRepository negotiationRepository;
 
   @Value("${negotiator.frontend-url}")
   private String frontendUrl;
 
+  @Value("${reminder.trigger-duration-days:P14D}")
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java:46]: thinking about it now, change to 7 days

File: src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java [hunk 3]
```diff
@@ -51,6 +75,29 @@ private static Set<Person> getRepresentativesForNegotiation(Negotiation negotiat
         .collect(Collectors.toSet());
   }
 
+  private static Set<Person> getRepresentativesForStaleNegotiation(Negotiation negotiation) {
+    Set<Person> representatives =
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java:79]: first you should just get resources involved in the Negotiation that did not have a new record in the last week/2 week. Then simply get their reps

File: src/test/java/eu/bbmri_eric/negotiator/integration/service/UserNotificationServiceTest.java [hunk 4]
```diff
@@ -278,4 +273,60 @@ void notifyUsersForNewPost_createdByRep_requesterIsNotified() {
     assertFalse(
         notificationRepository.findByRecipientId(negotiation.getCreatedBy().getId()).isEmpty());
   }
+
+  @Test
+  @WithMockNegotiatorUser(id = 109L, roles = "ADMIN")
+  public void testSendReminderEmails() {
```
Comment (RadovanTomik) [inline src/test/java/eu/bbmri_eric/negotiator/integration/service/UserNotificationServiceTest.java:279]: Not sure one test is enough for this. You should check multiple scenarios: for example set trigger duration to 0ms and then check if correct notifications were made

File: src/main/java/eu/bbmri_eric/negotiator/database/model/NegotiationResourceLifecycleRecord.java [hunk 5]
```diff
@@ -31,6 +31,10 @@ public class NegotiationResourceLifecycleRecord extends AuditEntity {
   @JoinColumn(name = "resource_id", referencedColumnName = "id")
   private Resource resource;
 
+  @ManyToOne
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/database/model/NegotiationResourceLifecycleRecord.java:34]: not sure this is needed. Just go through the parent entity, if this is needed you need a migration script no?

**Review by RadovanTomik — CHANGES_REQUESTED** [review]
Please see commments

**Review by RadovanTomik — APPROVED** [review]

**Review by konradlang — APPROVED** [review]

**Pull request merged**